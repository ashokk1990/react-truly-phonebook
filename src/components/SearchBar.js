import PropTypes from 'prop-types';
import React from "react";
class SearchBar extends React.PureComponent {
  state = {
    term: ""
  };

  handleSearchSubmit = event => {
    event.preventDefault();
    this.props.handleSubmit(this.state.term);
  };

  handleOnChange = event => {
    if (this.state.term && event.target.value === "") this.props.handleSubmit("");
    this.setState({ term: event.target.value });
  };

  render() {
    return (
      <div className="search-bar ui segment">
        <form className="ui form" onSubmit={this.handleSearchSubmit}>
          <div className="field">
            <label>Name Search Filter</label>
            <input
              type="text"
              value={this.state.term}
              onChange={this.handleOnChange}
            />
          </div>
        </form>
      </div>
    );
  }
}

SearchBar.propTypes = {
  handleSubmit: PropTypes.func.isRequired
}

export default SearchBar;
