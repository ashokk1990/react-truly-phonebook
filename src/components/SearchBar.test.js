import { shallow } from 'enzyme';
import React from "react";
import SearchBar from "./SearchBar";

const mockhandleSubmit = jest.fn();

const props = {
  handleSubmit: mockhandleSubmit,
};

describe('SearchBar Component', () => {
  let searchComponent = shallow(<SearchBar {...props} />);
  it('snapshot', () => {
    expect(searchComponent).toMatchSnapshot();
  });
  it('handleOnChange for blank text', () => {
    searchComponent.instance().handleOnChange({ target: { value: '' } });
    expect(searchComponent.state().term).toEqual('');
  });
  it('handleOnChange for correct text ', () => {
    searchComponent.instance().handleOnChange({ target: { value: 'Hello' } });
    expect(searchComponent.state().term).toEqual('Hello');
  });

  it('should call handleSubmit method when button is clicked', () => {
    const event = {preventDefault: () => {}}
    searchComponent.find('form').prop('onSubmit')(event);
    expect(mockhandleSubmit).toHaveBeenCalled();
  });
});
