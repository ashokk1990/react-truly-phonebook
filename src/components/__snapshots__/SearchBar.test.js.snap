// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component snapshot 1`] = `
<div
  className="search-bar ui segment"
>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        Name Search Filter
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
